### PR TITLE
Split core and typeface api

### PR DESCRIPTION
**Pull request by zTrap**

### CODE REVIEW

File: library-core/src/main/java/com/mikepenz/iconics/Iconics.kt [hunk 1]
```diff
@@ -61,16 +62,10 @@ object Iconics {
      */
     @JvmStatic fun init(context: Context? = null, fields: Array<Field>? = null) {
         if (context != null) {
-            if (!::applicationContext.isInitialized) {
-                this.applicationContext = context.applicationContext
-            }
+            IconicsContextHolder.applicationContext = context
```
Comment (mikepenz) [inline library-core/src/main/java/com/mikepenz/iconics/Iconics.kt:65]: the context given here may not be the application context.

File: library-core/src/main/java/com/mikepenz/iconics/Iconics.kt [hunk 2]
```diff
@@ -61,16 +62,10 @@ object Iconics {
...
         }
 
         if (!INIT_DONE) {
-            if (!::applicationContext.isInitialized) {
```
Comment (mikepenz) [inline library-core/src/main/java/com/mikepenz/iconics/Iconics.kt:70]: I think we should regardless have a null check here with the appropriate message

Comment (mikepenz) [reply library-core/src/main/java/com/mikepenz/iconics/Iconics.kt:70]: oh I see you moved this. :D

**Comment by mikepenz** [comment]
@zTrap may you add a few more details why you'd propose this change?

**Comment by zTrap** [comment]
In some libraries (such as String-Generator, or another) requires only typeface api. Also this can prevent redundant builds for typeface libraries when core changed

**Comment by mikepenz** [comment]
I see, but is the effort really this significant in those cases? compared to maintaining another module on the long term?

**Comment by zTrap** [comment]
Actually maintaining one more library is more hardly than have one complex. But this api must to be changed very rarely (I hope). Also in architecture it’s more right to separate api and implementations

File: library-core/src/main/java/com/mikepenz/iconics/Iconics.kt (hunk 1 above)
Comment (zTrap) [reply library-core/src/main/java/com/mikepenz/iconics/Iconics.kt:65]: `IconicsContextHolder` extract `ApplicationContext` from `Context` under the hood

**Pull request merged**